java/arcs/core/storage/handle/CollectionImpl.kt

```diff
@@ -27,27 +30,22 @@ private typealias SetHandle<T> = Handle<CrdtSet.Data<T>, CrdtSet.IOperation<T>,
 class CollectionImpl<T : Referencable>(
```

Comment:
in ts this T is not Referenceable, rather it takes entity instances that are serialized to make sure they have an ID. do we have something like that here?

Reply:
No idea-  I think jibbl might have the context on that but he's out sick. @jasonwyatt do you know?

Reply:
@jasonwyatt probably has better context. The use of `T : Referencable` here is done to match the type requirements on underlying CRDT type, which I believe was a decision made by him. 

Also possibly of note: these are the runtime-side handles; it may be the case that the SDK-side handles will have a slightly different shape.

Reply:
That's right, the Storage/CRDT stack is referencables-all-the-way-down - so to ensure things play nicely in the strongly typed world, all levels use Referencable as a lower bound on their type parameter.

Reply:
Ok, but there must be a place where the ID is added (if missing), right? Where is that done in this stack?